tests/unit/package_managers/yarn/test_resolver.py

```diff
@@ -0,0 +1,232 @@
+import json
+from typing import Any
+from unittest import mock
+
+import pytest
+
+from cachi2.core.errors import UnsupportedFeature
+from cachi2.core.package_managers.yarn.locators import parse_locator
+from cachi2.core.package_managers.yarn.resolver import Package, resolve_packages
+from cachi2.core.rooted_path import RootedPath
+
+
+def mock_yarn_info_output(yarn_info_outputs: list[dict[str, Any]]) -> str:
+    yarn_info_string_output = "\n".join(
+        json.dumps(obj, separators=(",", ":")) for obj in yarn_info_outputs
+    )
+    return yarn_info_string_output + "\n"
+
+
+# re-generate using hack/mock-unittest-data/yarn.sh
```

Comment:
Do we want to manually fix expected data once they are changed? Would it make sense to make this update part of generate-test-data in Makefile?

Reply:
IMO not really.
- `make generate-test-data` generates the expected output for integration tests.
- `hack/mock-unittest-data/yarn.sh` and `hack/mock-unittest-data/gomod.sh` generate the _input_ for _unit_ tests

There should rarely be a need to update those inputs, but the scripts are there in case you need them

Reply:
OK agree, we are instructed in a comment that such script to update data is included here, it's enough. :)